Tasks/TwineAuthenticateV1/twineauthenticatemain.ts

```diff
@@ -47,91 +52,136 @@ async function main(): Promise<void> {
 
         let pypircPath = utils.getPypircPath();
 
-        // create new file. We do not merge existing files and always create a fresh file
-        if (!tl.getVariable("PYPIRC_PATH") || !tl.exist(tl.getVariable("PYPIRC_PATH"))) {
-            fs.writeFileSync(pypircPath, formPypircFormatFromData(newEndpointsToAdd));
+        // create new .pypirc file if one does not exist yet
+        if (!tl.getVariable("PYPIRC_PATH") || !tl.exist(tl.getVariable("PYPIRC_PATH")) || !tl.exist(pypircPath)) {
+            fs.writeFileSync(pypircPath, '', 'utf8');
             tl.setVariable("PYPIRC_PATH", pypircPath, false);
             tl.debug(tl.loc("VariableSetForPypirc", pypircPath));
         }
-        else {
-            pypircPath = tl.getVariable("PYPIRC_PATH");
-            const pypirc = fs.readFileSync(pypircPath, 'utf8');
-            let fileContent = ini.parse(pypirc);
+        //else {
+        const pypirc = fs.readFileSync(pypircPath, 'utf8');
 
-            let usedRepos = new Set<string>();
+        let fileContent = ini.parse(pypirc);
 
-            for (let connection in fileContent) {
+        let usedRepos = new Set<string>();
 
-                const connectionObj: object = fileContent[connection];
+        for (let connection in fileContent) {
 
-                if (!connectionObj.hasOwnProperty('repository')) {
-                    const authenticatedRepo = getNestedRepoProperty(connectionObj);
+            const connectionObj: object = fileContent[connection];
 
-                    if (authenticatedRepo === undefined) {
-                        tl.warning(tl.loc("NoRepoFound", connection));
-                        continue;
-                    }
+            if (!connectionObj.hasOwnProperty('repository')) {
+                const authenticatedRepo = getNestedRepoProperty(connectionObj);
 
-                    usedRepos.add(authenticatedRepo);
+                if ((authenticatedRepo === undefined) && (connection.toLocaleLowerCase() !== 'distutils')) {
+                    tl.warning(tl.loc("NoRepoFound", connection));
                     continue;
                 }
 
-                usedRepos.add(connectionObj['repository']);
+                usedRepos.add(authenticatedRepo);
+                continue;
             }
 
-            let reposList: string[] = [];
+            usedRepos.add(connectionObj['repository']);
+        }
 
-            for (let entry of newEndpointsToAdd) {
+#if WIF
+        const entraWifServiceConnectionName = tl.getInput("workloadIdentityServiceConnection");
+        const feedUrl = tl.getInput("feedUrl");
 
-                tl.debug(tl.loc("Info_AddingAuthForRegistry", entry.packageSource.feedName));
```

Comment:
why not keep the auth registry debug line here?

Reply:
having it before at the top of the for loop could make debugging easier

Reply:
I had moved it to after the duplicate warning so that we would only print this message when we were about to authenticate the registry. I moved it back now, though